src/main/java/com/github/devotedmc/hiddenore/Config.java

```diff
@@ -47,23 +51,49 @@ private Config() {
 		ignoreSilktouch = false;
 		defaultPrefix = "You found hidden ore!";
 		transformAttemptMultiplier = 3;
+		configurations = new HashMap<String, FileConfiguration>();
 	}
 
 	public static void loadConfig() {
 		try {
-			file = HiddenOre.getPlugin().getConfig();
-			doLoad();
+			//TODO default-world.yml is never generated, but is required to generate per-world configs.
+			mainConfig = HiddenOre.getPlugin().getConfig();
+			doLoadMainConfig(mainConfig);
+			for (World world : HiddenOre.getPlugin().getServer().getWorlds()) {
+				File saveFile = new File(HiddenOre.getPlugin().getDataFolder(), String.format("%s-config.yml", world.getName()));
+				if (saveFile.exists()) {
+					//TODO pretty sure this whole scope here doesn't work.
+					//TODO double check if this is supposed to be the specific config or default-world.yml for the reader. I'm too tired to be able to follow this around in my head right now.
+					HiddenOre.getPlugin().getLogger().info("Loading configs for world " + world.getName());
+					Reader worldExistsReader = new InputStreamReader(HiddenOre.getPlugin().getResource(String.format("%s-config.yml", world.getName())));
+					configurations.put(world.getName(), YamlConfiguration.loadConfiguration(worldExistsReader));
+					//make sure that you put a file there first so it doesnt null pointer or just fix the routing dude
+					//configurations.get(world.getName()).load(saveFile);
+					HiddenOre.getPlugin().getLogger().info("Loaded Configs for world " + world.getName());
+				}else{
+					HiddenOre.getPlugin().getLogger().info("Configs do not exist for world " + world.getName() + ", generating them...");
+					File defaultWorldFile = new File(HiddenOre.getPlugin().getDataFolder(), "default-world.yml");
```

Comment:
In the same vein, this stuff here kind of does the exact opposite of what you might expect. It checks for the literal file-system existence of the "jar resource" default-world.yml. ... so it's probably not going to work.

Try instead:

```
try (Reader defaultWorldReader = new InputStreamReader(HiddenOre.getPlugin().getResource("default-world.yml"));) {
    configurations.put(world.getName(), YamlConfiguration.loadConfiguration(defaultWorldReader));
    configurations.get(world.getName()).save(new File(HiddenOre.getPlugin().getDataFolder(), String.format("%s-config.yml", world.getName())));
} catch (IOException ioe) {
    HiddenOre.getPlugin().getLogger().warning(String.format("A YAML file could not be created for the world named %s, default-world.yml missing from HiddenOre jar.", world.getName()));
}
```

bonus points: the try-with pattern above will close the resource when done ... I recommend using it in other places where applicable.